website/addons/dataverse/client.py

```diff
@@ -1,89 +1,95 @@
+        return None
 
 
 def connect_from_settings(user_settings):
-    return connect(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+    try:
+        return _connect(user_settings.api_token) if user_settings else None
+    except UnauthorizedError:
+        return None
 
 
-def connect_or_403(username, password, host=settings.HOST):
-    connection = Connection(
-        username=username,
-        password=password,
-        host=host,
-        disable_ssl=not settings.VERIFY_SSL,
-    )
-    if connection.status == http.FORBIDDEN:
-        raise HTTPError(http.FORBIDDEN)
-    return connection if connection.connected else None
+def connect_or_401(token):
+    try:
+        return _connect(token)
+    except UnauthorizedError:
+        raise HTTPError(http.UNAUTHORIZED)
 
 
-def connect_from_settings_or_403(user_settings):
-    return connect_or_403(
-        user_settings.dataverse_username,
-        user_settings.dataverse_password
-    ) if user_settings else None
+def connect_from_settings_or_401(user_settings):
+    return connect_or_401(user_settings.api_token) if user_settings else None
 
 
 def delete_file(file):
-    study = file.study
-    study.delete_file(file)
+    dataset = file.dataset
+    dataset.delete_file(file)
 
 
-def upload_file(study, filename, content):
-    study.upload_file(filename, content)
+def upload_file(dataset, filename, content):
+    dataset.upload_file(filename, content)
 
 
-def get_file(study, filename, released=False):
-    return study.get_file(filename, released)
+def get_file(dataset, filename, published=False):
+    return dataset.get_file(filename, published)
 
 
-def get_file_by_id(study, file_id, released=False):
-    return study.get_file_by_id(file_id, released)
+def get_file_by_id(dataset, file_id, published=False):
+    return dataset.get_file_by_id(file_id, published)
 
 
-def get_files(study, released=False):
-    return study.get_files(released)
+def get_files(dataset, published=False):
+    return dataset.get_files(published)
 
 
-def release_study(study):
-    return study.release()
+def publish_dataverse(dataverse):
+    try:
+        dataverse.publish()
+    except OperationFailedError:
+        raise HTTPError(http.BAD_REQUEST)
+
+
+def publish_dataset(dataset):
+    if dataset.get_state() == 'RELEASED':
+        raise HTTPError(http.CONFLICT)
+    if not dataset.dataverse.is_published:
+        raise HTTPError(http.METHOD_NOT_ALLOWED)
+
+    try:
+        dataset.publish()
+    except OperationFailedError:
+        raise HTTPError(http.BAD_REQUEST)
 
 
-def get_studies(dataverse):
+def get_datasets(dataverse):
     if dataverse is None:
         return [], []
-    accessible_studies = []
-    bad_studies = []    # Currently none, but we may filter some out
-    for s in dataverse.get_studies():
-        accessible_studies.append(s)
-    return accessible_studies, bad_studies
+    accessible_datasets = []
```

Comment:
What's the purpose of building this list? Is `accessible_datasets == dataverse.get_datasets()` or `list(dataverse.get_datasets())`?